core/src/cluster_info.rs

```diff
@@ -286,9 +292,15 @@ impl ClusterInfo {
     }
 
     pub fn push_vote(&mut self, vote: Transaction) {
+        if self.vote_cache.len() == MAX_CACHED_VOTES {
```

Comment:
Maybe while len >= MAX remove(0)?
If push vote is ever called from two threads, it can go over MAX votes, and will never recover